test/server_rendered_html_test.rb

```diff
@@ -51,4 +51,9 @@ def wait_to_ensure_asset_pipeline_detects_changes
     assert_no_match(/console.warn/, response.body)
     assert_no_match(/console.error/, response.body)
   end
+
+  test 'react inline component rendering' do
+    get '/server/inline_component'
+    assert_match(/Render this inline/, response.body)
```

Comment:
Added assertions for the tag, prerendering (`Render this inline</span>` instead of just `Render this inline`) and the layout (`<title>Dummy</title>`).